File: string_2/CatDog.py
'''
Return True if the string "cat" and "dog" appear the same number of times in
the given string.
cat_dog('catdog') → True
cat_dog('catcat') → False
cat_dog('1cat1cadodog') → True

@author LZ-FSDev
@see https://codingbat.com/prob/p164876
@since 3.10.2
@version 0.0.1
'''

def cat_dog(str):
    dog = 0
    cat = 0
    for i in range(len(str)-2):
        if (str[i:i+3] == "dog"):
            dog = dog + 1
        elif (str[i:i+3] == "cat"):
            cat = cat + 1
    return cat == dog

print("cat_dog('catdog') -> " + str(cat_dog('catdog')))
print("cat_dog('catcat') -> " + str(cat_dog('catcat')))
print("cat_dog('1cat1cadodog') -> " + str(cat_dog('1cat1cadodog')))

File: string_2/CatDog.java
/**
 * Return true if the string "cat" and "dog" appear the same number of times in
 * the given string.
 * 
 * <ul>
 *  <li>catDog("catdog") → true
 *  <li>catDog("catcat") → false
 *  <li>catDog("1cat1cadodog") → true
 * </ul>
 * 
 * @author LZ-FSDev
 * @see https://codingbat.com/prob/p111624
 * @since 17.0.1
 * @version 0.0.1
 */
public class CatDog {
    public static void main(String[] args) {
        System.out.println("catDog(\"catdog\") -> " + catDog("catdog"));
        System.out.println("catDog(\"catcat\") -> " + catDog("catcat"));
        System.out.println("catDog(\"1cat1cadodog\") -> " + catDog("1cat1cadodog"));
    }

    /**
     * Return true if the string "cat" and "dog" appear the same number of
     * times in the given string.
     * 
     * @param str A String.
     * @return true if the number of times "cat" and "dog" appear in 'str' the
     * same amount of times.
     * @since 0.0.1
     */
    public static boolean catDog(String str) {
        int cat = 0;
        int dog = 0;
        for (int i = 0; i < str.length()-2; i++) {
            if (str.substring(i, i+3).equals("cat")) {
                cat++;
            } else if (str.substring(i, i+3).equals("dog")) {
                dog++;
            }
        }
        System.out.println(cat);
        System.out.println(dog);
        return dog == cat;
    }    
}
